cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -617,9 +618,18 @@ public void markApplicationsLatest(Collection<ApplicationId> appIds)
   @Override
   public int addLatestApplication(ApplicationId id, ApplicationMeta meta) throws ConflictException {
     return TransactionRunners.run(transactionRunner, context -> {
+      ApplicationReference appRef = id.getAppReference();
+      RepositorySourceControlMetadataStore repoStore = getRepoSourceControlMetadataStore(context);
       getNamespaceSourceControlMetadataStore(context).write(id.getAppReference(),
           meta.getSourceControlMeta() == null ? SourceControlMeta.createDefaultMeta()
               : SourceControlMeta.builder(meta.getSourceControlMeta()).setSyncStatus(true).build());
+      if (repoStore.get(appRef) != null) {
```

Comment:
Should we not also write when `meta.getSourceControlMeta() != null` ?

Reply:
Sure, I have changed to -

```
if (meta.getSourceControlMeta() != null) {
          Instant lastSyncedAt = meta.getSourceControlMeta().getLastSyncedAt();
          repoStore.write(appRef, true, lastSyncedAt == null ? null :
              lastSyncedAt.toEpochMilli());
        }
        else{
          ImmutablePair<Long, Boolean> lastSyncedAndStatusPair = repoStore.get(appRef);
          if(lastSyncedAndStatusPair != null) {
            repoStore.write(appRef, false, lastSyncedAndStatusPair.getFirst());
          }
        }
```

Three cases should be considered:
1. If we are deploying an app in namespace and hasn't been pulled or pushed before - In this case we check, scm meta is null then we dont write anything in repo table since it is not present in the repo table.
2. If we are deploying an app already been pulled/pushed before - In this case the record will be present in repo table hence we will change the sync status but retain the last modified.
3. If we are pulling and deploying - In this case since scm meta will not be null we will be setting the values in repo table.

I feel the previous implementation would have also worked based on the concept that if we are pulling and deploying or are deploying an app which has been pulled/pushed before, the record would be there in the repo table.